lib/dep_selector/dependency_graph.rb

```diff
@@ -60,11 +60,17 @@ def generate_gecode_wrapper_constraints(packages_to_include_in_solve=nil)
             packages.map{ |name, pkg| pkg }
           end
 
+        logId = SecureRandom.uuid
         debugFlag = DebugOptionFile && File::exists?(DebugOptionFile)
+        Debug.log.level = Logger::INFO unless debugFlag
+        Debug.log.formatter = proc do |severity, datetime, progname, msg|
+          "#{logId}: #{msg}\n"
+        end
```

Comment:
I had basically that approach earlier - however, for some reason, setting the formatter proc from within the debug class does not work.  It does not error out, but it doesn't actually change the log format either.